vllm/config/__init__.py

```diff
@@ -194,7 +194,17 @@ def pairwise(iterable):
             yield a, b
             a = b
 
-    cls_node = ast.parse(textwrap.dedent(inspect.getsource(cls))).body[0]
+    try:
+        cls_node = ast.parse(textwrap.dedent(inspect.getsource(cls))).body[0]
+    except (OSError, KeyError, TypeError):
+        # getsource() fails in Python 3.13+ - use dataclass introspection
+        if is_dataclass(cls):
+            docs = {}
+            for field_obj in fields(cls):
+                docs[field_obj.name] = f"Configuration for {field_obj.name}."
```

Comment:
Is there a way to actually get the docstring for the field? otherwise it is not the same functionality?

Reply:
Yep, https://github.com/vllm-project/vllm/blob/007dd90859cc0337510536677418a43d8f66e286/vllm/config/__init__.py#L172

Reply:
Yeah this would nuke the CLI help strings

Reply:
Ok the issue is that the pydantic dataclass doesn't have a `__firstlineno__`.

They've fixed it in https://github.com/pydantic/pydantic/pull/11829/files#diff-8ff2ca642bcd607c31ac848396deaf4353b8b4a835473cbca58a22376fa72efa but haven't released yet

Reply:
You caught me :) I updated this with a real implementation, PTAL